impl/sql/mutationstorage/mutation_logs.go

```diff
@@ -153,23 +153,25 @@ func (m *Mutations) send(ctx context.Context, ts time.Time, directoryID string,
 		}
 	}()
 
-	var maxTime int64
+	var maxTime sql.NullTime
 	if err := tx.QueryRowContext(ctx,
-		`SELECT COALESCE(MAX(Time), 0) FROM Queue WHERE DirectoryID = ? AND LogID = ?;`,
+		`SELECT MAX(Time) FROM Queue WHERE DirectoryID = ? AND LogID = ?;`,
 		directoryID, logID).Scan(&maxTime); err != nil {
 		return status.Errorf(codes.Internal, "could not find max timestamp: %v", err)
 	}
-	tsTime := ts.UnixNano()
-	if tsTime <= maxTime {
+
+	// The Timestamp column has a maximum fidelity of microseconds.
+	// See https://dev.mysql.com/doc/refman/8.0/en/fractional-seconds.html
+	ts = ts.Truncate(time.Microsecond)
+	if ts.Before(maxTime.Time) || ts.Equal(maxTime.Time) {
 		return status.Errorf(codes.Aborted,
-			"current timestamp: %v, want > max-timestamp of queued mutations: %v",
-			tsTime, maxTime)
+			"current timestamp: %v, want > max-timestamp of queued mutations: %v", ts, maxTime)
```

Comment:
This logically preserves the previous behaviour, but is there any issue here with multiple writes in parallel? Constraints like this may be better enforced by the DB schema.